IdentityCore/src/cache/mac/MSIDMacCredentialStorageItem.m

```diff
@@ -126,7 +131,23 @@ - (instancetype)initWithJSONDictionary:(NSDictionary *)json
                 
                 if (credential)
                 {
-                    [instance.cacheObjects setObject:credential forKey:credentialKey];
+                    MSIDDefaultCredentialCacheKey *key = [self getDefaultCredentialCacheKey:credential];
+                    if (key)
+                    {
+                        MSIDCacheKey *storedCredentialKey = [self getStoredCredentialKey:key];
```

Comment:
what's the difference between storedCredentialKey and key? can we create "stored key" immediately?

Reply:
As discussed, created helper methods to create MSIDCacheKey directly from MSIDCredentialCacheItem and to create MSIDCacheKey from MSIDDefaultCredentialCacheKey.